### PR TITLE
Create common attributes object to handle complex schemas

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Simplify the definition of telemetry schemas to consolidate the appending of session properties, or one-off custom properties, if appropriate.

## Testing

Add unit tests and leverage existing tests to ensure integration works.

### CODE REVIEW

**Comment by bidetofevil** [comment]
* **#677** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/677?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bidetofevil and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private val sessionProperties: EmbraceSessionProperties? = null,
+    private val customAttributes: Map<String, String>? = null
+) {
+    private val map: MutableMap<EmbraceAttributeKey, String> = mutableMapOf()
+
+    /**
+     * Return a snapshot of the current values of the attributes set on this as a [Map]
+     */
+    fun snapshot(): Map<String, String> =
+        mutableMapOf<String, String>().plus(customAttributes ?: emptyMap())
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt:19]: Why is this `mutableMapOf` necessary?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt [hunk 2]
```diff
@@ -35,7 +35,12 @@ internal sealed class EmbType(type: String, subtype: String?) : TelemetryType {
 
         internal object Breadcrumb : System("breadcrumb")
 
-        internal object Log : System("log")
+        internal object Log : System("log") {
+            /**
+             * Attribute name for a unique id identifying the log
+             */
+            val embLogId = EmbraceAttributeKey("log_id")
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt:42]: Is this necessary? What would it look like in production & has this been discussed with backend folks? My understanding is that we can use the inbuilt IDs generated by OTel and that our own ID/unique name isn't required for logs as long as we add an `emb.type`

Comment (bidetofevil) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt:42]: This is just what was being written previously. @lucaslabari?

 If it's not necessary and can be removed, I can do that now or separately

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt (hunk 1 above)
Comment (bidetofevil) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt:19]: haha, it's not. left over from when I was doing `mapKeysTo` so I was merging into an existing map. I will correct.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt (hunk 2 above)
Comment (lucaslabari) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt:42]: Let me validate with backend and see if it's necessary.

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package io.embrace.android.embracesdk.arch.schema
+
+import io.embrace.android.embracesdk.internal.spans.toEmbraceAttributeName
+import io.embrace.android.embracesdk.session.properties.EmbraceSessionProperties
+
+/**
+ * Object that aggregates various attributes and returns a [Map] that represents the values at the current state
+ */
+internal class TelemetryAttributes(
```
Comment (lucaslabari) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt:9]: Will this be used by spans or just logs?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt (hunk 2 above)
Comment (lucaslabari) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt:42]: What about the other attributes?

Comment (bidetofevil) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/EmbType.kt:44]: They are common attributes that non-embrace logs will use as well so I put them in EmbraceAttributeKeys.kt

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt (hunk 3 above)
Comment (bidetofevil) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt:10]: it is only used by Logs now but can be used by spans and span events too. After this merges, I'll convert the other ones

**Review by lucaslabari — APPROVED** [review]
LGTM

**Pull request merged**